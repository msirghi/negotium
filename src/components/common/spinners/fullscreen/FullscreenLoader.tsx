import { makeStyles } from '@mui/styles';
import Loader from 'react-loader-spinner';
import { appTheme } from '../../../../common/theme/appTheme';
import { Box } from '@mui/system';

const useStyles = makeStyles({
  container: {
    width: '100vw',
    height: '100vh',
  },
  spinner: {
    position: 'absolute',
    top: '48%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
  },
});

export const FullscreenLoader = () => {
  const classes = useStyles();

  return (
    <Box className={classes.container}>
      <div className={classes.spinner}>
        <Loader
          type="Watch"
          color={appTheme.palette.primary.main}
          height={100}
          width={100}
        />
      </div>
    </Box>
  );
};
