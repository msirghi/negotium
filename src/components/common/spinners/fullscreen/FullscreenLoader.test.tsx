import renderer from 'react-test-renderer';
import { FullscreenLoader } from './FullscreenLoader';

describe('Fullscreen loader', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<FullscreenLoader />);
    expect(tree).toMatchSnapshot();
  });
});
