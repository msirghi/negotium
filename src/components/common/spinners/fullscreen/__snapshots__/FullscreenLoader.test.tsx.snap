// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fullscreen loader should match the snapshot 1`] = `
<div
  className="makeStyles-container-1 MuiBox-root css-0"
>
  <div
    className="makeStyles-spinner-2"
  >
    <div
      aria-busy="true"
      className=""
      style={Object {}}
    >
      <svg
        aria-label="audio-loading"
        enableBackground="new 0 0 100 100"
        height={100}
        id="L2"
        version="1.1"
        viewBox="0 0 100 100"
        width={100}
        x="0px"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <circle
          cx="50"
          cy="50"
          fill="none"
          r={48}
          stroke="#5886ff"
          strokeMiterlimit="10"
          strokeWidth="4"
        />
        <line
          fill="none"
          stroke="#5886ff"
          strokeLinecap="round"
          strokeMiterlimit="10"
          strokeWidth="4"
          x1="50"
          x2="85"
          y1="50"
          y2="50.5"
        >
          <animateTransform
            attributeName="transform"
            dur="2s"
            from="0 50 50"
            repeatCount="indefinite"
            to="360 50 50"
            type="rotate"
          />
        </line>
        <line
          fill="none"
          stroke="#5886ff"
          strokeLinecap="round"
          strokeMiterlimit="10"
          strokeWidth="4"
          x1="50"
          x2="49.5"
          y1="50"
          y2="74"
        >
          <animateTransform
            attributeName="transform"
            dur="15s"
            from="0 50 50"
            repeatCount="indefinite"
            to="360 50 50"
            type="rotate"
          />
        </line>
      </svg>
    </div>
  </div>
</div>
`;
